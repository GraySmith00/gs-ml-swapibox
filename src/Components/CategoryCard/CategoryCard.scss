.category-card {
  display: flex;
  flex-direction: column;

  padding-bottom: 2em;

  border-radius: 5px;
  color: #020202;
  background-color: rgba(250, 250, 250, 0.9);

  -webkit-box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 62px -22px rgba(0, 0, 0, 0.75);

  &:hover {
    box-shadow: 0 10px 20px 0 rgba(0, 0, 0, 0.6);
    transform: translateY(-1px);
  }
}

i {
  margin: 0.5em 0;
}

i.star.far.fa-star::before {
  cursor: pointer;
}

.card-image {
  border-radius: 5px 5px 0 0;
  height: 200px;
  width: 100%;
}

p,
i,
h3 {
  margin: 0.5em 2em;
}
