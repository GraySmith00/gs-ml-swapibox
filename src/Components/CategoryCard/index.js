import React from 'react';
import PropTypes from 'prop-types';

const CategoryCard = ({ item, toggleFavorite }) => {
  const cards = Object.keys(item).map((itemKey, index) => {
    switch (itemKey) {
      case 'name':
        return <h3 key={`${itemKey}-${index}`}>{item[itemKey]}</h3>;
      case 'favorite':
        return (
          <i
            onClick={() => toggleFavorite(item)}
            className={`star ${item.favorite ? 'fas fa-start' : 'far fa-star'}`}
            key={`${itemKey}-${index}`}
          />
        );
      default:
        return (
          <p key={`${itemKey}-${index}`}>{`${itemKey}: ${item[itemKey]}`}</p>
        );
    }
  });
  return <div>{cards}</div>;
};

CategoryCard.propTypes = {
  item: PropTypes.object.isRequired,
  toggleFavorite: PropTypes.func.isRequired
};

export default CategoryCard;

// {
//   name,
//   planet,
//   population,
//   species,
//   favorite,
//   climate,
//   residents,
//   terrain,
//   model,
//   passengers,
//   vehicle_class
// }
