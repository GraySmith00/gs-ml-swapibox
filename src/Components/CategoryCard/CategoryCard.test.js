import React from 'react';
import { shallow } from 'enzyme';

import CategoryCard from '../CategoryCard';

describe('CategoryCard component', () => {
  let wrapper;
  let mockItem;

  beforeEach(() => {
    mockItem = {
      name: 'Luke Skywalker',
      planet: 'Tatooine',
      population: '200000',
      species: 'Human',
      favorite: false
    };
    wrapper = shallow(<CategoryCard item={mockItem} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
