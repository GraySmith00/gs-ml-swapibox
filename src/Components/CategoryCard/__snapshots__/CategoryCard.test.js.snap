// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard component should match the snapshot 1`] = `"<div><h3>Luke Skywalker</h3><p>planet: Tatooine</p><p>population: 200000</p><p>species: Human</p><i class=\\"star far fa-star\\"></i></div>"`;
